### PR TITLE
feat: add jbang-vscode to codium + add usability vscode settings

### PR DESCRIPTION
**Pull request by maxandersen**
jbang-vscode is still in preview mode but it is too good to not be enabled imo.

plus i realized vscodium defaults are slightly bad so added a few default settings
such as autosave on, zoom by trackpad/mousewheel, linked editing and method name guesing.

### CODE REVIEW

File: src/main/java/dev/jbang/cli/Edit.java [hunk 1]
```diff
@@ -226,12 +226,39 @@ private static void setupEditor(Path editorBinPath, Path dataPath) throws IOExce
...
 
+		Path settingsjson = dataPath.resolve("user-data/User/settings.json");
+
+		if (!Files.exists(settingsjson)) {
+			verboseMsg("Setting up some good default settings at " + settingsjson);
+			Files.createDirectories(settingsjson.getParent());
+
+			String vscodeSettings = "{\n" +
+			// better than breadcrumbs
+					"    \"editor.experimental.stickyScroll.enabled\": true,\n" +
```
Comment (fbricon) [inline src/main/java/dev/jbang/cli/Edit.java:237]: doesn't work well with Java code. Try a file with a copyright header, or annotations

**Comment by fbricon** [comment]
will it still generate the eclipse project files?

**Comment by maxandersen** [comment]
> will it still generate the eclipse project files?

for now yes; but my thinking is that in futureif you do jbnag edit <folder> it_wont, but for jbang edit xyz.java it will continue to do the linked trick as it works more places out of box.

**Comment by fbricon** [comment]
that means if you open a jbang script it'll first be imported as an eclipse project. Dependency validation will require clicking on the Synchronize JBang code lens

**Comment by maxandersen** [comment]
yes, but users using codium from command line will enjoy it.

**Pull request merged**